sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -38,15 +38,15 @@
  *
  * The `numElements` is 8 bytes storing the number of elements of this array.
  *
- * In the `null bits` region, we store 1 bit per element, represents whether a element has null
- * Its total size is ceil(numElements / 8) bytes, and  it is aligned to 8-byte boundaries.
+ * In the `null bits` region, we store 1 bit per element, represents whether an element is null
+ * Its total size is ceil(numElements / 8) bytes, and it is aligned to 8-byte boundaries.
  *
  * In the `values or offset&length` region, we store the content of elements. For fields that hold
  * fixed-length primitive types, such as long, double, or int, we store the value directly
- * in the field. For fields with non-primitive or variable-length values, we store a relative
- * offset (w.r.t. the base address of the array) that points to the beginning of
- * the variable-length field and length (they are combined into a long). For variable length
- * portion, each is aligned to 8-byte boundaries.
+ * in the field. For fixed-length portion, each is word-aligned. For fields with non-primitive or
```

Comment:
`each is word-aligned` this is confusing, sounds like each element value(even for byte) is word-aligned.